modules/CbSystem/main/systemImpl.cpp

```diff
@@ -176,6 +180,60 @@ void systemImpl::check_update_marker() {
     }
 }
 
+void systemImpl::setSystemTime(const std::chrono::time_point<date::utc_clock>& timepoint) {
+    // we ignore:
+    // - enabling or checking of NTP support
+    // - checking the time difference
+    // we do only:
+    // - set the system time with the simplest and most straight-forward method
+    //   (SDBus)
+    // - throw on errors
+    __attribute__((cleanup(sd_bus_error_free))) sd_bus_error sd_error = SD_BUS_ERROR_NULL;
+    __attribute__((cleanup(sd_bus_unrefp))) sd_bus* bus = NULL;
+    const char* bus_timedate_destination = "org.freedesktop.timedate1";
+    const char* bus_timedate_path = "/org/freedesktop/timedate1";
+    const char* bus_timedate_interface = "org.freedesktop.timedate1";
+    int rv;
+
+    // date::utc_clock takes leap seconds into consideration, the SDBus system clock does not
+    // so convert to eliminate to 28 seconds difference
+    const auto timepoint_sys_clock = date::clock_cast<std::chrono::system_clock>(timepoint);
+    const auto timepoint_sys_clock_us = std::chrono::time_point_cast<std::chrono::microseconds>(timepoint_sys_clock);
+    const int64_t usec_utc_sys = timepoint_sys_clock_us.time_since_epoch().count();
+
+    rv = sd_bus_default_system(&bus);
+    if (rv < 0) {
+        sd_bus_close(bus);
```

Comment:
I would expect that sd_bus_default_system() closes bus in error case, so this is not necessary.